src/implementations/twig/components/button/button.story.js

```diff
@@ -11,117 +9,162 @@ import dataCall from '@ecl/specs-component-button/demo/data--call';
 import dataGhost from '@ecl/specs-component-button/demo/data--ghost';
 import dataSearch from '@ecl/specs-component-button/demo/data--search';
 
+import defaultSprite from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
 import button from './button.html.twig';
 import notes from './README.md';
 
 const iconsList = {};
-iconsList.none = '';
-
 uiIcons.forEach((icon) => {
   iconsList[icon] = icon;
 });
 
-// Preserve the adapted specs.
-const prepareButton = (data, type) => {
-  const typeLabel = type ? tabLabels.required : tabLabels.optional;
-  data.disabled = boolean('disabled', data.disabled, tabLabels.states);
-  data.label = text('label', data.label, tabLabels.required);
-
-  data.variant = select(
-    'variant',
-    [data.variant],
-    data.variant,
-    tabLabels.required
-  );
-
-  data.type = select('type', [data.type], data.type, typeLabel);
-
-  getExtraKnobs(data);
+const prepareControls = (data, args) => {
+  data.label = args.label;
+  data.disabled = args.disabled;
+  if (args.icon_name != null) {
+    data.icon = {};
+    data.icon.name = args.icon_name;
+    data.icon.type = 'ui';
+    data.icon.size = 'xs';
+    data.icon.transform = args.icon_transform;
+    data.icon.path = defaultSprite;
+    data.icon_position = args.icon_position;
+  } else if (args.icon_name == null && data.icon) {
+    delete data.icon;
+  }
 
   return data;
 };
 
 export default {
   title: 'Components/Button',
-  decorators: [withKnobs, withCode, withNotes],
+  argTypes: {
+    label: {
+      name: 'label',
+      type: { name: 'string', required: true },
+      defaultValue: '',
+      description: 'The main label of the button',
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '' },
```

Comment:
Do we need empty summary items if nothing to put in them?

Reply:
yes it showcase an empty string in the table